docs/lang/concurrency.md

```diff
@@ -0,0 +1,112 @@
+Concurrency Extension
+=====================
+
+This extension brings *lightweight parallel threads* to Bril.
+
+*  **`spawn`** launches a new thread and returns an opaque handle.
+*  **`join`** blocks until a thread identified by such a handle finishes.
+*  **`thread`** is a new primitive type used for those handles.
+
+Threads share *heap memory* (the region allocated by the memory
+extension) but **do not share local variables or SSA temporaries**.
+Data races on the shared heap are *undefined behaviour*.
+
+---
+
+Primitive Type
+--------------
+
+| Type    | Meaning                                       |
+|---------|-----------------------------------------------|
+| thread  | Opaque handle returned by `spawn`, consumed by `join`. |
+
+---
+
+Instructions
+------------
+
+### `spawn`
+tid: thread = spawn @func arg1 arg2 … argN
+
+JSON form:
+```jsonc
+{
+  "op":   "spawn",
+  "dest": "tid",
+  "type": "thread",
+  "funcs": ["func"],
+  "args":  ["arg1", "arg2", "..."]
+}
+
+* Creates a new thread that begins executing function @func.
```

Comment:
Markdown backticks would be helpful here, and in other places where you refer to parts of the instruction:
```suggestion
* Creates a new thread that begins executing function `@func`.
```